### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by Theodus**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,12 +3,12 @@ config ?= release
 PACKAGE := peg
 GET_DEPENDENCIES_WIITH := corral fetch
 CLEAN_DEPENDENCIES_WIITH := corral clean
-COMPILE_WITH := corral run -- ponyc
+COMPILE_WITH := corral run -- ponyc -V1
 
 BUILD_DIR ?= build/$(config)
 SRC_DIR := $(PACKAGE)
 EXAMPLES_DIR := examples
-tests_binary := $(BUILD_DIR)/$(PACKAGE)
```
Comment (SeanTAllen) [inline Makefile]: why this change? it would make the peg library different than our others. I'd prefer to keep them the same unless there's a strong reason for the change.

Comment (Theodus) [reply Makefile]: Ok, I've reverted that bit

File: Makefile [hunk 2]
```diff
@@ -31,9 +31,9 @@ test: unit-tests build-examples
 unit-tests: $(tests_binary)
 	$^ --exclude=integration --sequential
 
-$(tests_binary): $(SOURCE_FILES) | $(BUILD_DIR)
+$(tests_binary): $(wildcard test/*.pony) $(SOURCE_FILES) | $(BUILD_DIR)
```
Comment (SeanTAllen) [inline Makefile]: Given this is a library, we could move the test related things into peg/test and keep this the same as the other libraries.

File: Makefile [hunk 3]
```diff
@@ -31,9 +31,9 @@ test: unit-tests build-examples
...
 	${GET_DEPENDENCIES_WIITH}
-	$(PONYC) -o ${BUILD_DIR} $(SRC_DIR)
+	$(PONYC) -o ${BUILD_DIR} -b $(PACKAGE) test
```
Comment (SeanTAllen) [inline Makefile]: With the above suggested change, I think this could be left the same as well. Assuming we are ok with the above change.

If we are going to do tests in the top level for libraries, I'd prefer to do it across all of them.

File: Makefile (hunk 2 above)
Comment (Theodus) [reply Makefile]: I've reverted the Makefile changes and placed the tests in `peg/_test.pony`

**Review by SeanTAllen — APPROVED** [review]
